### DISCUSSION TITLE
How to to intercept parametrized sql string and how to suppress inserting to database

### DISCUSSION THREAD

**Discussion by yvef** [nhibernate/nhibernate-core#2785] [Q&A]
Hi,
I have two question.
How can I correctly intercept prepared sql string? I use OnPrepareStatement method of IInterceptor, but it provides non parametrized string like this: "INSERT INTO MYTABLE (COL1, COL2, COL3) VALUES ("?", "?", "?")".
And also, is there any way to prevent database interaction? As far as I see, it is possible to create a custom driver. But how can this be registered? 

Thank you a lot.

**Comment by victorgutmann** [comment]
I'm also interested in the answer to the first question. I need access to the parameter values passed to SQL, just like Nhibernate logs do by placing the values at the end of sql.
